modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -486,27 +485,24 @@ public ObjectField updateCustomObjectField(
 			List<ObjectFieldSetting> objectFieldSettings)
 		throws PortalException {
 
+		_validateLabel(labelMap);
```

Comment:
Why did you choose to move this validation up?

Reply:
To fix an SF error. 

Without the `if (objectDefinition.isSystem())`, I had to leave all the code that depends on the `newObjectField` variable right after initializing it.

So we can move this validation down (after `newObjectField.setLabelMap`) or up. I decided to move it up because I think it improves performance (fail early).

Reply:
It makes sense to me.